code/go/0chain.net/smartcontract/zcnsc/mint_test.go

```diff
@@ -81,6 +81,45 @@ func Test_EmptySignaturesShouldFail(t *testing.T) {
 	require.Error(t, err)
 }
 
+func Test_EmptyAuthorizersShouldFail(t *testing.T) {
+	ctx := MakeMockStateContext()
+	ctx.stakingPools = nil
+	ctx.authorizers = nil
+
+	contract := CreateZCNSmartContract()
+	payload, err := CreateMintPayload(ctx, defaultClient)
+	require.NoError(t, err)
+
+	transaction, err := CreateTransaction(defaultClient, "mint", payload.Encode(), ctx)
+	require.NoError(t, err)
+
+	_, err = contract.Mint(transaction, payload.Encode(), ctx)
+	require.Error(t, err)
+}
+
+func Test_EmptyAuthorizersNonemptySignaturesShouldFail(t *testing.T) {
+	ctx := MakeMockStateContext()
+	ctx.stakingPools = nil
+	ctx.authorizers = nil
+
+	contract := CreateZCNSmartContract()
+	payload, err := CreateMintPayload(ctx, defaultClient)
+	require.NoError(t, err)
+
+	// Add a few signatures.
+	var signatures []*AuthorizerSignature
+	for i := 0; i < 10; i++ {
+		signatures = append(signatures, &AuthorizerSignature{})
+	}
+	payload.Signatures = signatures
+
+	transaction, err := CreateTransaction(defaultClient, "mint", payload.Encode(), ctx)
+	require.NoError(t, err)
+
+	_, err = contract.Mint(transaction, payload.Encode(), ctx)
```

Comment:
The `numAuth` is not empty in the `Mint` contract, the error is returned when verifying signatures, please check the place where the error is returned and make sure the tests do test the code we want.

Reply:
Fixed

Reply:
The error will still be "signatures not found" because since numAuth is zero so signatures will be sliced to an empty array.

https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/zcnsc/mint.go#L62